### PR TITLE
Inventory - Cache inventory postInit config lookups

### PR DESCRIPTION
**Pull request by dedmen**
A picture says more than a thousand words:

### CODE REVIEW

File: addons/fastroping/CfgWeapons.hpp [hunk 1]
```diff
@@ -1,60 +1,60 @@
...
+        GVAR(ropeLength) = 36.6;
+        picture = QPATHTOF(data\m_rope_ca);
+        model = "\A3\Structures_F_Heli\Items\Tools\Rope_01_F.p3d";
+        displayName = CSTRING(Rope_36_Display);
+        descriptionShort = CSTRING(descriptionShort);
+        class ItemInfo: CBA_MiscItem_ItemInfo {
+            mass = 108;
+        };
+    };
+};
```
Comment (commy2) [inline addons/fastroping/CfgWeapons.hpp]: What happens in this file?

File: addons/inventory/XEH_postInit.sqf [hunk 2]
```diff
@@ -8,28 +8,10 @@ if (!hasInterface) exitWith {};
...
-private _allItems = [];
-
-_allItems append ("getNumber (_x >> 'scope') > 0" configClasses (configFile >> "CfgWeapons"));
-_allItems append ("getNumber (_x >> 'scope') > 0" configClasses (configFile >> "CfgGlasses"));
-_allItems append ("getNumber (_x >> 'scope') == 2" configClasses (configFile >> "CfgMagazines"));
-_allItems append ("getNumber (_x >> 'scope') > 0 && {getNumber (_x >> 'isBackpack') == 1}" configClasses (configFile >> "CfgVehicles"));
-
-{_x call _fnc_addToCache; false} count _allItems;
+//isEqualType is hacking protection as we cannot trust that the cache hasn't been manipulated
+{GVAR(ItemKeyNamespace) setVariable _x;} forEach (_allItems select {_x isEqualType []});
```
Comment (commy2) [inline addons/inventory/XEH_postInit.sqf]:
```suggestion
{
    if (_x isEqualType []) then {
        GVAR(ItemKeyNamespace) setVariable _x;
    };
} forEach _allItems;
```

File: addons/inventory/XEH_postInit.sqf [hunk 3]
```diff
@@ -8,28 +8,10 @@ if (!hasInterface) exitWith {};
 // but if the item has the same name and picture it at least shouldn't change the filter anyway
 // luckily we don't need private items, so dummy and parent classes are out of the picture
 
-GVAR(ItemKeyNamespace) = [] call CBA_fnc_createNamespace;
```
Comment (commy2) [inline addons/inventory/XEH_postInit.sqf]: This line is still needed, no?

File: addons/fastroping/CfgWeapons.hpp (hunk 1 above)
Comment (dedmen) [reply addons/fastroping/CfgWeapons.hpp]: Huh? I didn't... wtf..

File: addons/inventory/XEH_postInit.sqf [hunk 4]
```diff
@@ -9,46 +9,22 @@ if (!hasInterface) exitWith {};
...
-
-private _allItems = [];
-
-_allItems append ("getNumber (_x >> 'scope') > 0" configClasses (configFile >> "CfgWeapons"));
-_allItems append ("getNumber (_x >> 'scope') > 0" configClasses (configFile >> "CfgGlasses"));
-_allItems append ("getNumber (_x >> 'scope') == 2" configClasses (configFile >> "CfgMagazines"));
-_allItems append ("getNumber (_x >> 'scope') > 0 && {getNumber (_x >> 'isBackpack') == 1}" configClasses (configFile >> "CfgVehicles"));
-
-{_x call _fnc_addToCache; false} count _allItems;
+} forEach ([[], _allItems] select (_allItems isEqualType []));
```
Comment (commy2) [inline addons/inventory/XEH_postInit.sqf]:
```suggestion
} forEach _allItems;
```

File: addons/inventory/XEH_postInit.sqf [hunk 5]
```diff
@@ -9,46 +9,22 @@ if (!hasInterface) exitWith {};
...
+private _allItems = uiNamespace getVariable [QGVAR(ItemKeyCache), []]; //See XEH_preStart.sqf
 
-private _fnc_addToCache = {
-    private _displayName = getText (_this >> "displayName");
-    private _picture = getText (_this >> "picture");
-
-    // list box seems to delete the leading backslash
-    if (_picture select [0,1] == "\") then {
-        _picture = _picture select [1];
+//isEqualType is hacking protection as we cannot trust that the cache hasn't been manipulated
```
Comment (commy2) [inline addons/inventory/XEH_postInit.sqf]:
```suggestion
// isEqualType is hacking protection as we cannot trust that the cache hasn't been manipulated
if !(_allItems isEqualType []) then {
    _allItems = [];
};
```

File: addons/inventory/XEH_postInit.sqf [hunk 6]
```diff
@@ -9,46 +9,22 @@ if (!hasInterface) exitWith {};
...
+{
+    if (_x isEqualType []) then {
```
Comment (commy2) [inline addons/inventory/XEH_postInit.sqf]:
```suggestion
    if (_x isEqualType [] && {_x isEqualTypeArray ["", configNull]}) then {
```

File: addons/inventory/XEH_preStart.sqf [hunk 7]
```diff
@@ -1,3 +1,59 @@
...
+
+    // list box seems to delete the leading backslash
+    if (_picture select [0,1] == "\") then {
+        _picture = _picture select [1];
+    };
+
+    [format ["%1:%2", _displayName, _picture], _x];
+}];
+
+
```
Comment (commy2) [inline addons/inventory/XEH_preStart.sqf]:
```suggestion
```

File: addons/inventory/XEH_preStart.sqf [hunk 8]
```diff
@@ -1,3 +1,59 @@
 #include "script_component.hpp"
 
 #include "XEH_PREP.hpp"
+
+
```
Comment (commy2) [inline addons/inventory/XEH_preStart.sqf]:
```suggestion
```

File: addons/inventory/XEH_postInit.sqf (hunk 4 above)
Comment (dedmen) [reply addons/inventory/XEH_postInit.sqf]: Why?

File: addons/inventory/XEH_postInit.sqf (hunk 6 above)
Comment (dedmen) [reply addons/inventory/XEH_postInit.sqf]: I found that this thing is the new major performance waste in this function.. 
Maybe compileFinal will be better in the end.

Comment (commy2) [reply addons/inventory/XEH_postInit.sqf]: Sounds simpler anyway.

Comment (dedmen) [reply addons/inventory/XEH_postInit.sqf]: Nvm doesn't work.. Would need to re-resolve all the config entries which would be even worse

**Pull request merged**